src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -214,22 +216,28 @@ public static SupportedPlatformData GetSupportedPlatforms(CompletionItem item, W
             return null;
         }
 
-        public static int GetContextPosition(CompletionItem item)
+        public static async Task<int> GetContextPositionAsync(
+            Document document, CompletionItem item, CancellationToken cancellationToken)
         {
-            if (item.Properties.TryGetValue("ContextPosition", out var text) && int.TryParse(text, out var number))
+            if (item.Properties.TryGetValue("ContextPosition", out var text) &&
+                int.TryParse(text, out var number))
             {
-                return number;
+                // We have no access to the editor at this layer.  So it's not 
+                // possible for us to map the original context position forward
+                // to the current position in the file.  So we need to cap the
+                // positoin to make sure it's within the bounds of the current
+                // text.
```

Comment:
Is this the right fix? It seems to me that the problem is that CompletionItem is not associated with a snapshot of the document it was calculated for.